src/common/settingsProvider.ts

```diff
@@ -34,21 +33,26 @@ export class SettingsProvider {
     return networkEnabled;
   }
 
-  // Legacy only: this function returns the theme for the legacy bundled DevTools
-  getThemeSettings(): ThemeString {
-    const settings = vscode.workspace.getConfiguration(SETTINGS_STORE_NAME);
-    const themeString: ThemeString = settings.get('themes') || 'System preference';
-    return themeString;
-  }
-
   // This function returns the theme for the new frame hosted DevTools by:
   // 1. Fetching the User configured Global VSCode theme, return it if supported
   // 2. Fall back to the extension Theme setting selector (light, dark, system preference)
   // 3. Fall back to system preference
   getThemeFromUserSetting(): string {
       const themeSetting = vscode.workspace.getConfiguration().get('workbench.colorTheme') as string;
-      const legacySetting = (vscode.workspace.getConfiguration(SETTINGS_STORE_NAME).get('themes') || '') as string;
-      return SUPPORTED_THEMES.get(themeSetting) || SUPPORTED_THEMES.get(legacySetting) || 'systemPreferred';
+      let theme = SUPPORTED_THEMES.get(themeSetting);
+      if (!theme) {
+        switch (vscode.window.activeColorTheme.kind as number) {
```

Comment:
casting this as a number since the ColorThemeKind enum only goes up to 3 even though there are now 4 values.

See https://github.com/microsoft/vscode/blob/c20b68a62d198e73df34de5f95671eb63c8b5ff2/src/vs/workbench/api/common/extHostTypes.ts#L3091